staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go

```diff
@@ -864,6 +939,13 @@ func (alloc *allocator) allocateOne(r deviceIndices, allocateSubRequest bool) (b
 					deviceIndex:     r.deviceIndex + 1,
 				}
 				done, err := alloc.allocateOne(deviceKey, allocateSubRequest)
+				// If we hit the allocation size limit, don't attempt
+				// to find a different device, as it will not change the
+				// number of allocated devices.
+				if errors.Is(err, errAllocationResultMaxSizeExceeded) {
+					deallocate()
+					return false, err
+				}
```

Comment:
I'm confused again. Compared to the next line, this if branch only adds a `deallocate()`. `return false, err` is the same.

This could be combined to:
```
if err != nil {
    if errors.Is(err, errAllocationResultMaxSizeExceeded) {
        deallocate()
    }
    return false, err
}
```

But how does that match the comment?